File: Java/TP3_MET_JAVA/src/Iteratif/RepresentationIteratif.java
package Iteratif;

public class RepresentationIteratif implements VisiteurMotIteratif<String> {

    @Override
    public String casVide() {
        return "";
    }

    @Override
    public String casCaractereMot(Character c, String m) {
        return m + c;
    }
}

File: TypeScript/TP3_MET_TS/src/Iteratif/RepresentationIteratif.ts
import {VisiteurMotIteratif} from "./VisiteurMotIteratif";

export class RepresentationIteratif implements VisiteurMotIteratif<string> {

    public casVide(): string {
        return "";
    }

    public casCaractereMot(c: string, m: string): string {
        return m + c;
    }
}
